activerecord/Rakefile

```diff
@@ -207,52 +207,50 @@ end
 
 namespace :db do
   namespace :mysql do
-    mysql2_config = ARTest.config["connections"]["mysql2"]
-    mysql2_connection_arguments = lambda do |connection_name|
-      mysql2_connection = mysql2_config[connection_name]
-      ["--user=#{mysql2_connection["username"]}", ("--password=#{mysql2_connection["password"]}" if mysql2_connection["password"]), ("--host=#{mysql2_connection["host"]}" if mysql2_connection["host"]), ("--socket=#{mysql2_connection["socket"]}" if mysql2_connection["socket"])].join(" ")
+    mysql_configs = ARTest.config["connections"].slice("mysql2", "trilogy").transform_values { |connections| connections.slice("arunit", "arunit2") }
```

Comment:
Looks like `connector` and `connection_name` aren't used in your iterations below, so maybe we could just iterate the connection configs like before?

Reply:
The connection config includes more keys than `mysql2` and `trilogy` (`sqlite3` and `postgresql`) at the top level, and more keys than `arunit` and `arunit2` in the sub-level, stuff generated here:
https://github.com/rails/rails/blob/1717c0ef058c17c80e6e8d1bae14d6d18c8ea64b/activerecord/test/support/config.rb#L28-L29

These `slice` calls reduce the content of `mysql_configs` to what we need to setup the MySQL users and databases.

Reply:
```
mysql_configs = ARTest.config["connections"].values_at("mysql2", "trilogy").flat_map do |connections|
  connections.values_at("arunit", "arunit2")
end
```
Is what I had in mind, yes. Less noise in the loops.

Reply:
Will update. 👍🏻